src/tools/cgget.c

```diff
@@ -835,7 +844,12 @@ int main(int argc, char *argv[])
 
 #ifdef CGXGET
 	ret = convert_cgroups(&cg_list, cg_list_len, version, CGROUP_DISK);
-	if (ret)
+	if (ret == ECGNOVERSIONCONVERT && ignore_unmappable)
+		/* The user has specified that we should ignore any errors
+		 * due to being unable to map from v1 to v2 or vice versa
+		 */
+		ret = 0;
```

Comment:
`goto err;` is required here too, otherwise `get_value()` will try to read the non-existing control value and trigger an error ```variable file read failed No such file or directory```

Reply:
I don't think so.  The way I did it, `ignore_mappable` throws an error and doesn't populate that setting.  But there could be other valid settings that do need to be converted.

Unless I'm missing something entirely... :)

Reply:
Sorry for being slow on the uptake.  You are right - this is a problem because cgxget will display every setting if the `cgroup_controller struct` doesn't have any settings within it.

I've pushed a proposed fix.  Here are the [two](https://github.com/drakenclimber/libcgroup/commit/11f6606827872bcebfcd7a6e3a5391eff7ae2e33#diff-5e93f3166e4154f7d03bcbb4228ecbf5f10fee8e82d724d830c8f1a224061504R235) [key](https://github.com/drakenclimber/libcgroup/commit/11f6606827872bcebfcd7a6e3a5391eff7ae2e33#diff-5e93f3166e4154f7d03bcbb4228ecbf5f10fee8e82d724d830c8f1a224061504R288) parts of the proposed fix.  The tests all pass, but I'm not explicitly testing this case.  We should add this.

Reply:
After thinking more about it, this code won't work.  `cgc` will be a stale pointer to the freed memory.  Bummer
```
			if (!cgc)
				/*
				 * The converted controller had no settings
				 * and was removed.  It's up to us to manage
				 * the controller count
				 */
				out_cgroup->index--;
```

Reply:
Fixed now.  I passed `struct cgroup_controller ** const out_cgc` to `convert_controller()` and NULL out the the pointer if the controller is empty.

Reply:
I added a test for this explicit case.  Here's the code coverage showing it got hit on both a v1 and v2 system:
https://coveralls.io/builds/46171147/source?filename=src%2Fabstraction-common.c#L220